hs/rsh/stdlib.rsh

```diff
@@ -203,3 +203,70 @@ export const sqrt = (y, k) =>
       ? [ x, ((y / x + x) / 2) ]
       : [ z, x ]
   )[1];
+
+export const FixedPoint = Object({ scale: UInt, i: UInt });
+
+export const mk_fixed_point = (scale) => (i) =>
+  ({ scale, i });
+
+export const fixed_point_convert_scale = (x, scale) => {
+  assert(scale / x.scale >= 0,
+    "fixed_point_convert_scale: Scale division causes underflow.");
+  return (x.scale == scale)
+    ? x
+    : { i: (x.i * scale) / x.scale, scale };
+}
+
+export const fixed_point_homogenize_scales = (x, y) => {
+  const scale = x.scale < y.scale ? y.scale : x.scale;
+  const x_ = fixed_point_convert_scale(x, scale);
+  const y_ = fixed_point_convert_scale(y, scale);
+  return [ scale, x_, y_ ];
+}
+
+export const fixed_point_add = (x, y) => {
+  const [ scale, x_, y_ ] = fixed_point_homogenize_scales(x, y);
+  assert(x_.i + y_.i <= UInt.max, "fixed_point_add: Addition causes overflow.");
```

Comment:
Why is this needed... given that `verifyOverflow` will add this when necessary?

Reply:
Oh duh, I'll remove these `assert`s